owid/catalog/meta.py

```diff
@@ -144,6 +143,27 @@ def version(self) -> Optional[str]:
 
         return None
 
+    def update_from_yaml(self, path: Union[Path, str]) -> None:
+        """The main reason for wanting to do this is to manually override what goes into Grapher before an export."""
+        with open(path) as istream:
+            annot = yaml.safe_load(istream)
+
+        # update sources of dataset
+        for source_annot in annot["dataset"].get("sources", []):
+            # if there's an existing source, update it
+            ds_sources = [s for s in self.sources if s.name == source_annot["name"]]
+            if ds_sources:
+                for k, v in source_annot.items():
+                    setattr(ds_sources[0], k, v)
+            # otherwise create new source
+            else:
+                self.sources.append(Source(**source_annot))
```

Comment:
`Source` also has a `from_dict` method, though this is probably equivalent. No need to change.